sdk/storage/storage-blob/review/storage-blob.api.md

```diff
@@ -1345,6 +1345,7 @@ export interface BlockBlobParallelUploadOptions extends CommonOptions {
     };
     onProgress?: (progress: TransferProgressEvent) => void;
     tags?: Tags;
+    tier?: BlockBlobTier | string;
```

Comment:
Do you want the "| string" here? Can tier be any string? If you're trying to allow using string literals and enums, then the best approach is to have the signature be the string literal union of allowed values and offer the enum as well. IOW, this works fine:

```
enum Animal { Cat = "Cat", Dog = "Dog" };
function foo(a: "Cat" | "Dog") { };
foo(Animal.Cat);
foo("Cat");
```

Reply:
I find some comments here: https://github.com/jeremymeng/azure-sdk-for-js/commit/355726654df03075900bd9286e5670d05d63bef9
Though it's refactoring so not sure about the original author. I guess we did it for future compatibility in case new AccessTiers are supported. So will follow this pattern for now.
Adding @jeremymeng @XiaoningLiu @HarshaNalluru for more context.

```ts
export function toAccessTier(
  tier: BlockBlobTier | PremiumPageBlobTier | string | undefined
): Models.AccessTier | undefined {
  if (tier == undefined) {
    return undefined;
  }

  return tier as Models.AccessTier; // No more check if string is a valid AccessTier, and left this to underlay logic to decide(service).
}
```